examples/preprocessing/muscle_ica.py

```diff
@@ -0,0 +1,118 @@
+
+"""
+# Authors: Alex Rockhill <aprockhill@mailbox.org>
+#
+# License: BSD-3-Clause
+
+# %%
+
+import os.path as op
+import mne
+
+data_path = mne.datasets.sample.data_path()
+raw_fname = op.join(data_path, 'MEG', 'sample', 'sample_audvis_raw.fif')
+raw = mne.io.read_raw_fif(raw_fname)
+raw.crop(tmin=100, tmax=130)  # take 30 seconds for speed
+
+# pick only EEG channels, muscle artifact is basically not picked up by MEG
+# if you have a simulatenous recording, you may want to do ICA on MEG and EEG
+# separately
+raw.pick_types(eeg=True)
+
+# ICA works best with a highpass filter applied
+raw.load_data()
+raw.filter(l_freq=1., h_freq=None)
+
+# %%
+# Run ICA
+ica = mne.preprocessing.ICA(
+    n_components=15, method='picard', max_iter='auto', random_state=97)
+ica.fit(raw)
+
+# %%
+# Remove components with postural muscle artifact using ICA
+ica.plot_sources(raw)
+
+# %%
+# By inspection, let's select out the muscle-artifact components based on
+# :footcite:`DharmapraniEtAl2016` manually.
+#
+# The criteria are:
+#   - Positive slope of log-log power spectrum between 7 and 75 Hz
+#     (here just flat because it's not in log-log)
+#   - Peripheral focus or dipole/multi-pole foci (the blue and red
+#     blobs in the topomap are far from the vertex where the most
+#     muscle is)
+#   - Single focal point (low spatial smoothness; there is just one focus
+#     of the topomap compared to components like the first ones that are
+#     more likely neural which spread across the topomap)
+#
+# The other attribute worth noting is that the time course in
+# :func:`mne.preprocessing.ICA.plot_sources` looks like EMG; you can
+# see spikes when each motor unit fires so that the time course looks fuzzy
+# and sometimes has large spikes that are often at regular intervals.
+#
+# ICA component 13 is a textbook example of what muscle artifact looks like.
+# The focus of the topomap for this component is right on the temporalis
+# muscle near the ears. There is also a minimum in the power spectrum at around
+# 10 Hz, then a maximum at around 25 Hz, then a slow linear dropoff (relative)
+# to non-muscle components) in non-log-log units (this slope is positive in
+# log-log units) this is a very typical pattern for muscle artifact.
+
+muscle_idx = [6, 7, 8, 9, 10, 11, 12, 13, 14]
+ica.plot_properties(raw, picks=muscle_idx)
+
+# first, remove blinks and heartbeat to compare
+blink_idx = [0]
+heartbeat_idx = [5]
+ica.apply(raw, exclude=blink_idx + heartbeat_idx)
+ica.plot_overlay(raw, exclude=muscle_idx)
+
+# %%
+# Finally, let's try an automated algorithm to find muscle components
+# and ensure that it gets the same components we did manually.
+muscle_idx_auto, scores = ica.find_bads_muscle(raw)
+ica.plot_scores(scores)
+print(f'Manually found muscle artifact ICA components:      {muscle_idx}\n'
+      f'Automatically found muscle artifact ICA components: {muscle_idx_auto}')
+
+# %%
+# Compare to EEGBCI dataset
+
+muscle_idxs = {1: [9], 2: [5, 10, 12, 13, 14]}
```

Comment:
you have a threshold parameter in find_bads_muscle. I would suggest you use the muscle_idx_auto to exclude some components and not pass manually the components.